controllers/spacerequest/spacerequest_controller.go

```diff
@@ -43,12 +63,440 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, memberClusters map[strin
+
+	case len(subSpaces.Items) == 1:
+		// deleting subSpace
+		return r.deleteExistingSubSpace(logger, &subSpaces.Items[0])
+
+	default:
+		// something went wrong and there are too many subspaces
+		return false, fmt.Errorf("invalid number of subSpaces found. actual %d, expected %d", len(subSpaces.Items), 1)
+	}
+
+}
+
+// deleteExistingSubSpace deletes a given space object in case deletion was not issued already.
+// returns true/nil if the deletion of the space was triggered
+// returns false/nil if the space was already deleted
+// return false/err if something went wrong
+func (r *Reconciler) deleteExistingSubSpace(logger logr.Logger, subSpace *toolchainv1alpha1.Space) (bool, error) {
+	if util.IsBeingDeleted(subSpace) {
+		logger.Info("the subSpace resource is already being deleted")
+		deletionTimestamp := subSpace.GetDeletionTimestamp()
+		if time.Since(deletionTimestamp.Time) > 120*time.Second {
+			return false, fmt.Errorf("subSpace deletion has not completed in over 2 minutes")
+		}
+		return true, nil // subspace is still being deleted
+	}
+
+	logger.Info("deleting the subSpace resource", "subSpace name", subSpace.Name)
+	// Delete subSpace associated with SpaceRequest
+	if err := r.Client.Delete(context.TODO(), subSpace); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil // was already deleted
+		}
+		return false, errs.Wrap(err, "unable to delete subspace") // something wrong happened
+	}
+	logger.Info("deleted the subSpace resource", "subSpace name", subSpace.Name)
+	return true, nil
+}
+
+func (r *Reconciler) setStatusTerminating(memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest) error {
+	return r.updateStatus(
+		spaceRequest,
+		memberCluster,
+		toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.ConditionReady,
+			Status: v1.ConditionFalse,
+			Reason: toolchainv1alpha1.SpaceTerminatingReason,
+		})
+}
 
-	return ctrl.Result{}, nil
+func (r *Reconciler) setStatusTerminatingFailed(logger logr.Logger, memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, cause error) error {
+	if err := r.updateStatus(
+		spaceRequest,
+		memberCluster,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.ConditionReady,
+			Status:  v1.ConditionFalse,
+			Reason:  toolchainv1alpha1.SpaceTerminatingFailedReason,
+			Message: cause.Error(),
+		}); err != nil {
+		logger.Error(cause, "unable to terminate Space")
+		return err
+	}
+	return cause
+}
+
+func (r *Reconciler) setStatusFailedToCreateSubSpace(logger logr.Logger, memberCluster cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, cause error) error {
+	if err := r.updateStatus(
+		spaceRequest,
+		memberCluster,
+		toolchainv1alpha1.Condition{
+			Type:    toolchainv1alpha1.ConditionReady,
+			Status:  v1.ConditionFalse,
+			Reason:  toolchainv1alpha1.SpaceProvisioningFailedReason,
+			Message: cause.Error(),
+		}); err != nil {
+		logger.Error(err, "unable to terminate Space")
```

Comment:
```suggestion
		logger.Error(err, "unable to create Space")
```

Reply:
applied in [60159bc](https://github.com/codeready-toolchain/host-operator/pull/770/commits/60159bc49fc4872acb0b880853cb3775e5454300)